builder/scaleway/step_snapshot.go

```diff
@@ -37,8 +38,15 @@ func (s *stepSnapshot) Run(ctx context.Context, state multistep.StateBag) multis
 		Zone:       scw.Zone(c.Zone),
 	}
 	timeout := c.SnapshotTimeout
-	if timeout != 0 {
-		waitSnapshotRequest.Timeout = scw.TimeDurationPtr(timeout)
+	duration, err := time.ParseDuration(timeout)
```

Comment:
Put the logic of timeout validation in the config. When you are using it, it should be defined in the config and used directly